Infusion/Packets/Client/TargetLocationRequest.cs

```diff
@@ -9,20 +9,21 @@ public TargetLocationRequest(CursorId cursorId, Location3D location, ModelId til
             var payload = new byte[19];
             var writer = new ArrayPacketWriter(payload);
 
-            writer.WriteByte((byte) PacketDefinitions.TargetCursor.Id);
-            writer.WriteByte((byte) CursorTarget.Location);
-            writer.WriteUInt(cursorId.Value);
-            writer.WriteByte((byte) cursorType);
-            writer.WriteInt(0); // clicked on item = 0 (using location click)
+            writer.WriteByte((byte)PacketDefinitions.TargetCursor.Id);
+            writer.WriteByte((byte)CursorTarget.Location);
+            writer.WriteUInt(cursorId.Value); //TARGET SERIAL
+            writer.WriteByte((byte)cursorType); //FLAG
+            writer.WriteInt(0); // ALSO TARGET SERIAL FOR OBJECT PACKETS
             writer.WriteUShort((ushort)location.X);
             writer.WriteUShort((ushort)location.Y);
-            writer.WriteByte(0); // unknown
-            writer.WriteSByte((sbyte)location.Z);
-            writer.WriteModelId(tileType);
+            writer.WriteUShort((ushort)location.Z);
+            writer.WriteUShort((ushort)tileType); 
+
```

Comment:
Please, what is the difference?

- Z coordinate is defined only by one byte, so if you cast it to ushort you get zero from the first byte and Z coordinate as sbyte from the second byte.
- Both `WriteUShort` and `WriteModelId` writes the same two bytes to the stream.